src/bindings/python/simgrid_python.cpp

```diff
@@ -164,6 +232,180 @@ PYBIND11_MODULE(simgrid, m)
+                        fun();
+                      } catch (const py::error_already_set& ex) {
+                        if (ex.matches(pyForcefulKillEx)) {
+                          XBT_VERB("Actor killed");
+                          simgrid::ForcefulKillException::do_throw(); // Forward that ForcefulKill exception
+                        }
+                        throw;
+                      }
+                    },
+                    hosts);
+              },
+              "Create an actor from a function or an object. See the :ref:`example <s4u_ex_actors_create>`.");
+  auto mpi_m =
+      smpi_m.def_submodule("MPI", "Bindings of MPI functions.")
+          .def(
+              "Init", []() { return MPI_Init(); }, py::call_guard<py::gil_scoped_release>(),
+              "Initialize the MPI interface")
+          .def(
+              "Init",
+              [](std::vector<std::string> args) {
+                auto argc = static_cast<int>(args.size());
+                std::vector<char*> argv(args.size() + 1); // argv[argc] is nullptr
+                std::transform(begin(args), end(args), begin(argv), [](std::string& s) { return &s.front(); });
+                // Currently this can be dangling, we should wrap this somehow.
+                char** argv_data = argv.data();
+                return MPI_Init(&argc, &argv_data);
+              },
+              "Initialize the MPI interface")
+          .def("Finalize", &MPI_Finalize, py::call_guard<py::gil_scoped_release>(), "Finalize the MPI interface");
+  py::class_<simgrid::smpi::Comm, std::unique_ptr<simgrid::smpi::Comm, py::nodelete>> mpi_comm(
+      mpi_m, "Comm", "Simulated host. See the C++ documentation for details.");
+
+  mpi_comm
+      .def(
+          "size",
+          [](simgrid::smpi::Comm* comm) {
+            int size;
+            MPI_Comm_size(comm, &size);
+            return size;
+          },
+          "Get the size of the communicator")
+      .def(
+          "rank",
+          [](simgrid::smpi::Comm* comm) {
+            int rank;
+            MPI_Comm_rank(comm, &rank);
+            return rank;
+          },
+          "Get the rank within the communicator");
+  mpi_comm.def_property_readonly_static("WORLD",
+                                        [](py::object /* self */) { return (simgrid::smpi::Comm*)MPI_COMM_WORLD; });
+  mpi_comm.def_property_readonly_static("SELF",
+                                        [](py::object /* self */) { return (simgrid::smpi::Comm*)MPI_COMM_SELF; });
+
+  mpi_m.def("Alltoall", wrap_mpi_alltoall<int>, py::call_guard<py::gil_scoped_release>(), "Alltoall operation");
+  mpi_m.def("Alltoall", wrap_mpi_alltoall<double>, py::call_guard<py::gil_scoped_release>(), "Alltoall operation");
+  mpi_m.def("Alltoall", wrap_mpi_alltoall<float>, py::call_guard<py::gil_scoped_release>(), "Alltoall operation");
+  mpi_m.def("Alltoall", wrap_returned_mpi_alltoall<int>, py::call_guard<py::gil_scoped_release>(),
```

Comment:
This code is getting somewhat repetitive :)

Reply:
have you a way to avoid the repetition here? I can use C macro. but do not feel the right way.